File: api/app/schemas/reports.py
from __future__ import annotations

from pydantic import BaseModel


class ReportApprove(BaseModel):
    analyst_notes: str | None = None


class ReportTechnicianSummary(BaseModel):
    id: str
    username: str
    full_name: str


class ReportTicketSummary(BaseModel):
    id: str
    title: str
    station_id: str
    anomaly_zone: str | None
    status: str | None = None


class InspectionReport(BaseModel):
    id: str
    ticket_id: str
    ticket: ReportTicketSummary | None
    technician_id: str
    technician: ReportTechnicianSummary | None
    notes: str | None
    severity: str | None
    root_cause: str | None
    corrective_action: str | None = None
    issue_resolved: bool | None = None
    submitted_at: str | None
    analyst_approved: bool
    analyst_approved_at: str | None
    analyst_notes: str | None
    round: int = 1
    is_active: bool = True
    created_at: str


class InspectionReportListResponse(BaseModel):
    pending: list[InspectionReport]
    follow_up: list[InspectionReport]
    approved: list[InspectionReport]

File: web/src/types/reports.ts
export interface ReportTechnicianSummary {
  id: string;
  username: string;
  full_name: string;
}

export interface ReportTicketSummary {
  id: string;
  title: string;
  station_id: string;
  anomaly_zone: string | null;
  status?: string | null;
}

export interface InspectionReport {
  id: string;
  ticket_id: string;
  ticket: ReportTicketSummary | null;
  technician_id: string;
  technician: ReportTechnicianSummary | null;
  notes: string | null;
  severity: "low" | "medium" | "high" | null;
  root_cause: string | null;
  corrective_action: string | null;
  issue_resolved: boolean | null;
  submitted_at: string | null;
  analyst_approved: boolean;
  analyst_approved_at: string | null;
  analyst_notes: string | null;
  round: number;
  is_active: boolean;
  created_at: string;
}

export interface InspectionReportListResponse {
  pending: InspectionReport[];
  follow_up: InspectionReport[];
  approved: InspectionReport[];
}

export interface ReportApprove {
  analyst_notes?: string;
}
